src/Participant.cpp

```diff
@@ -35,106 +37,113 @@ Participant::Participant()
     {
-        if (config["domain_id"])
+        dds_participant_->set_listener(nullptr);
+
+        if (fastrtps::types::ReturnCode_t::RETCODE_OK !=
+                ::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(dds_participant_))
         {
-            const ::fastdds::dds::DomainId_t domain_id = config["domain_id"].as<uint32_t>();
-            build_participant(domain_id);
+            logger_ << eprosima::is::utils::Logger::Level::ERROR
+                    << "Cannot delete Fast DDS participant yet: it has active entities" << std::endl;
         }
-        else
-        {
-            std::ostringstream err;
-            err << "The node 'participant' in the YAML configuration of the 'fastdds' system "
-                << "must be a map containing two keys: 'file_path' and 'profile_name'";
+    }
+}
 
-            throw DDSMiddlewareException(logger_, err.str());
-        }
+void Participant::build_participant(
+        const YAML::Node& config)
+{
+    // Set 0 as default domain
+    eprosima::fastdds::dds::DomainId_t domain_id(0);
 
+    // Check if domain_id tag is present in configuration, if not 0 as default
+    if (config["domain_id"])
+    {
+        domain_id = config["domain_id"].as<uint32_t>();
     }
-    else
+
+    logger_ << eprosima::is::utils::Logger::Level::DEBUG
+            << "Creating Fast DDS Participant in domain " << domain_id << std::endl;
+
+    // Load default XML files
+    fastrtps::xmlparser::XMLProfileManager::loadDefaultXMLFile();
+
+    // Loading XML if file_path is given
+    if (config["file_path"])
     {
-        const std::string file_path = config["file_path"].as<std::string>();
-        if (XMLP_ret::XML_OK != XMLProfileManager::loadXMLFile(file_path))
+        if (fastrtps::xmlparser::XMLP_ret::XML_OK !=
+            fastrtps::xmlparser::XMLProfileManager::loadXMLFile(config["file_path"].as<std::string>()))
         {
+            std::ostringstream err;
+            err << "Failed to load XML file provided in 'file_path': " << config["file_path"].as<std::string>()
+                << ". It cannot be found or is incorrect.";
             throw DDSMiddlewareException(
-                      logger_, "Loading provided XML file in 'file_path' field was not successful");
+                logger_, err.str());
         }
+    }
 
-        const std::string profile_name = config["profile_name"].as<std::string>();
+    // Debug variable
+    std::string participant_name;
 
-        dds_participant_ = this->create_participant_with_profile(profile_name);
+    // If profile_name is given in configuration, the other tags do not apply
+    if (!config["profile_name"])
+    {
+        // Participant QoS
+        ::fastdds::dds::DomainParticipantQos participant_qos;
 
-        if (dds_participant_)
+        // Depending the SH type, use participant qos or databroker qos
+        // TODO : change for databroker alias refactor
+        if (config["discovery-server"])
```

Comment:
This is to maintain consistency with the rest of the tags.

```suggestion
        if (config["discovery_server"])
```